lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -149,13 +178,36 @@ class RenderBlockingResources extends Audit {
    * @param {Node} fcpGraph
    * @param {Set<string>} deferredIds
    * @param {Map<string, number>} wastedCssBytesByUrl
+   * @param {LH.Artifacts.DetectedStack[]} Stacks
    * @return {number}
    */
-  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl) {
-    const originalEstimate = simulator.simulate(fcpGraph).timeInMs;
+  static estimateSavingsWithGraphs(simulator, fcpGraph, deferredIds, wastedCssBytesByUrl, Stacks) {
+    const {nodeTimings} = simulator.simulate(fcpGraph);
+    const stackSpecificTimings = new Map(nodeTimings);
 
     let totalChildNetworkBytes = 0;
     const minimalFCPGraph = /** @type {NetworkNode} */ (fcpGraph.cloneWithRelationships(node => {
+      if (nodeTimings) {
```

Comment:
for typedefs of APIs that *we* have written *in Lighthouse*, you can trust the types to be correct and not be defensive

if you're dealing with the typedefs of an API of another library that we might not trust as much then being defensive is a little more acceptable :)